sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/SparkMetadataOperationSuite.scala

```diff
@@ -333,4 +333,31 @@ class SparkMetadataOperationSuite extends HiveThriftJdbcTest {
       assert(pos === 17, "all columns should have been verified")
     }
   }
+
+  test("get columns operation should handle interval column properly") {
```

Comment:
Nit: Can we also test an extract query of the table with calendar types?

Reply:
This PR fixes SparkGetColumnsOperation. As a meta operation, it is not related to extract-queries that executed by SparkExecuteStatementOperation. BTW, I notice that test cases for extracting interval values in VIEWS through the thrift server might be missing in the current codebase. Maybe we can make another PR to improve these, WDYT, @cloud-fan Or I can just add some UTs here https://github.com/yaooqinn/spark/blob/d24d27f1bc39e915df23d65f8fda0d83e716b308/sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/HiveThriftServer2Suites.scala#L674

Reply:
That's why I think we should add a small check here to make sure they actually work.

Reply:
I'm OK to add the missing test in this PR.

Reply:
done, thank you guys for the check~